### PR TITLE
[storage] Adds from_string to models

### LINKED ISSUE #7805: [Storage-file] Apply "permissions" model changes to other models

### PR DESCRIPTION
**Pull request by kristapratico**
Resolves #7805

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-blob/HISTORY.md [hunk 1]
```diff
@@ -6,7 +6,8 @@
 
 - `set_container_access_policy` has required parameter `signed_identifiers`.
 - NoRetry policy has been removed. Use keyword argument `retry_total=0` for no retries. 
-
+- `AccountSasPermissions`, `BlobSasPermissions`, `ContainerSasPermissions`, `ResourceTypes`, and `Services`
```
Comment (rakshith91) [inline sdk/storage/azure-storage-blob/HISTORY.md]: The first three ones are already done in preview 4..You want to remove them from this.

File: sdk/storage/azure-storage-blob/azure/storage/blob/_shared/models.py [hunk 2]
```diff
@@ -249,32 +239,26 @@ class ResourceTypes(object):
...
                 ('o' if self.object else ''))
 
+    def __str__(self):
+        return self._str
+
+    @classmethod
+    def from_string(cls, string):
+        service = 's' in string
+        container = 'c' in string
+        object = 'o' in string
```
Comment (rakshith91) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_shared/models.py]: object is a reserved keyword in python...we dont want to redefine it...perhaps rename these to res_service, res_object etc?

File: sdk/storage/azure-storage-blob/azure/storage/blob/_shared/models.py [hunk 3]
```diff
@@ -361,33 +342,26 @@ class Services(object):
...
+        return self._str
 
-Services.BLOB = Services(blob=True)
-Services.QUEUE = Services(queue=True)
-Services.FILE = Services(file=True)
+    @classmethod
+    def from_string(cls, string):
+        blob = 'b' in string
+        queue = 'q' in string
+        file = 'f' in string
```
Comment (rakshith91) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_shared/models.py]: same here...file is a reserved keyword..

File: sdk/storage/azure-storage-file/azure/storage/file/_shared/models.py [hunk 4]
```diff
@@ -249,32 +239,26 @@ class ResourceTypes(object):
         A string representing the resource types.
```
Comment (rakshith91) [inline sdk/storage/azure-storage-file/azure/storage/file/_shared/models.py]: You want to remove the documentation for param str _str: everywhere.

**Pull request merged**